packages/clerk-js/src/core/auth/AuthCookieService.ts

```diff
@@ -162,7 +162,8 @@ export class AuthCookieService {
     }
 
     try {
-      const token = await this.clerk.session.getToken();
+      // Use refreshIfStale to fetch fresh token when cached token is within leeway period
+      const token = await this.clerk.session.getToken({ refreshIfStale: true });
```

Comment:
This also affects the `refreshTokenOnFocus`, but do we want that? Because those cases can be a race to finish setting the cookie before any external data fetching runs, it seems prudent to use the available token then?